### PR TITLE
Dataplane Learning for fvSubnet

### PR DESCRIPTION
**Pull request by mfr-6**
Hello Team,

This PR aims to add data plane learning argument to fvSubnet.

Please carry out the review, thank you.

### CODE REVIEW

File: aci/data_source_aci_fvsubnet.go [hunk 1]
```diff
@@ -61,7 +62,6 @@ func dataSourceAciSubnet() *schema.Resource {
 
 			"virtual": &schema.Schema{
 				Type:     schema.TypeString,
-				Optional: true,
```
Comment (akinross) [inline aci/data_source_aci_fvsubnet.go:64]: I see you have changed to computed for attributes in data source and in the documentation. Is there a reason to skip next_hop_addr, msnlb + nested attributes, and anycast_mac? These can also all be changed to computed from what I see.

File: website/docs/d/subnet.html.markdown [hunk 2]
```diff
@@ -37,18 +37,19 @@ data "aci_subnet" "dev_subnet" {
...
-- `name_alias` - (Optional) Name alias for object subnet.
-- `preferred` - (Optional) Indicates if the subnet is preferred (primary) over the available alternatives. Only one preferred subnet is allowed.
-- `scope` - (Optional) The List of network visibility of the subnet.
-- `virtual` - (Optional) Treated as virtual IP address. Used in case of BD extended to multiple sites.
-- `relation_fv_rs_bd_subnet_to_out` - (Optional) Relation to class l3extOut. Cardinality - N_TO_M. Type - Set of String.
-- `relation_fv_rs_nd_pfx_pol` - (Optional) Relation to class ndPfxPol. Cardinality - N_TO_ONE. Type - String.
-- `relation_fv_rs_bd_subnet_to_profile` - (Optional) Relation to class rtctrlProfile. Cardinality - N_TO_ONE. Type - String.
-- `next_hop_addr` - (Optional) EP Reachability of the Application EPGs Subnet object. Type - String.
-- `msnlb` - (Optional) A block representing MSNLB of the Application EPGs Subnet object. Type - Block.
+- `annotation` - (Read-Only) Annotation for object subnet.
```
Comment (akinross) [inline website/docs/d/subnet.html.markdown:40]: Could you add ` Type: String.` consistently behind all string inputs?

File: website/docs/d/subnet.html.markdown [hunk 3]
```diff
@@ -37,18 +37,19 @@ data "aci_subnet" "dev_subnet" {
...
+- `description` - (Read-Only) Description for object subnet.
+- `ctrl` - (Read-Only) The list of subnet control state. The control can be specific protocols applied to the subnet such as IGMP Snooping.
+- `ip_data_plane_learning` - (Read-Only) Flag to enable/disable ip-data-plane learning for the Subnet object. Allowed values are "enabled" and "disabled" and default value is "enabled". Type: String.
+- `name_alias` - (Read-Only) Name alias for object subnet.
+- `preferred` - (Read-Only) Indicates if the subnet is preferred (primary) over the available alternatives. Only one preferred subnet is allowed.
+- `scope` - (Read-Only) The List of network visibility of the subnet.
+- `virtual` - (Read-Only) Treated as virtual IP address. Used in case of BD extended to multiple sites.
+- `relation_fv_rs_bd_subnet_to_out` - (Read-Only) Relation to class l3extOut. Cardinality - N_TO_M. Type - Set of String.
```
Comment (akinross) [inline website/docs/d/subnet.html.markdown:48]:
```suggestion
- `relation_fv_rs_bd_subnet_to_out` - (Read-Only) Relation to class l3extOut. Cardinality - N_TO_M. Type: Set of String.
```

File: website/docs/r/subnet.html.markdown [hunk 4]
```diff
@@ -112,6 +123,7 @@ Class - fvSubnet
 - `annotation` - (Optional) Annotation for object subnet.
 - `description` - (Optional) Description for object subnet.
 - `ctrl` - (Optional) The list of subnet control state. The control can be specific protocols applied to the subnet such as IGMP Snooping. Allowed values are "unspecified", "querier", "nd" and "no-default-gateway". Default is "nd". NOTE: "unspecified" should't be used along with other values.
+- `ip_data_plane_learning` - (Optional) Flag to enable/disable ip-data-plane learning for the Subnet object. Allowed values are "enabled" and "disabled" and default value is "enabled". Type: String.
```
Comment (akinross) [inline website/docs/r/subnet.html.markdown:126]: Could you add  Type: String. consistently behind all string inputs?

File: website/docs/r/subnet.html.markdown [hunk 5]
```diff
@@ -23,86 +23,97 @@ Class - fvSubnet
...
-		description = "This subject is created by terraform"
-		ctrl        = ["no-default-gateway"]
-		preferred   = "no"
-		virtual     = "yes"
-		msnlb {
-			mode  = "mode-uc"
-			mac   = "00:1F:20:34:89:AA"
-		}
-	}
+resource "aci_subnet" "foosubnet" {
```
Comment (akinross) [inline website/docs/r/subnet.html.markdown:26]: Can you also update the subnet.tf example in the examples/subnet directory to include ip data plane learning?

**Review by akinross — CHANGES_REQUESTED** [review]

**Review by akinross — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Comment by lindie1337** [comment]
Thanks very much for this implementation!

**Pull request merged**